src/formats/notion/convert-to-md.ts

```diff
@@ -434,13 +437,62 @@ function splitBrsInFormatting(body: HTMLElement, tagName: FormatTagName) {
 	body.innerHTML = htmlString;
 }
 
-function replaceTableOfContents(body: HTMLElement) {
-	const tocLinks = body.findAll('a[href*=\\#]') as HTMLAnchorElement[];
-	for (const link of tocLinks) {
-		if (link.getAttribute('href')?.startsWith('#')) {
-			link.setAttribute('href', '#' + link.textContent);
-		}
+
+function getTOCIndent(tocItem: Element | null): Number {
+	return Number(tocItem?.classList[1].slice(-1) ?? -1);
+}
+
+function appendTOCItem(itemNew: Element, itemPre: Element | null | undefined) {
+	if (!itemPre) return;
```

Comment:
Why does this function accept null or undefined for `itemPre` if it's going to then return without appending in those cases?

Reply:
Good point. Originally, there was a possibility that `itemPre` could have be `undefined`, if recursing too far up the tree. However, the way `formatTableOfContents` is currently implemented, that should be impossible.

Reply:
I've removed the `null | undefined` now, as it is redundant.